storeys/js/utils/datastructures.js

```diff
@@ -0,0 +1,77 @@
+define(
+    ['require'],
+    function(require) {
+        /**
+         *A subclass of dictionary customized to handle multiple values for the
+         *same key.
+
+         *>>> var d = new MultiValueDict({'name': ['Adrian', 'Simon'], 'position': ['Developer']})
+         *>>> d['name']
+         *'Simon'
+         *>>> d.getlist('name')
+         *['Adrian', 'Simon']
+         *>>> d.getlist('doesnotexist')
+         *[]
+         *>>> d.getlist('doesnotexist', ['Adrian', 'Simon'])
+         *['Adrian', 'Simon']
+         *>>> d.get('lastname', 'nonexistent')
+         *'nonexistent'
+         *>>> d.setlist('lastname', ['Holovaty', 'Willison'])
+
+         *This class exists to solve the irritating problem raised by cgi.parse_qs,
+         *which returns a list for every key, even though most Web forms submit
+         *single name-value pairs.
+         */
+        function MultiValueDict(obj) {
```

Comment:
I didn't found structure like `hash_tables` inside javascript (simple {} - is unsuitable because it doesn't supports multiple values for same key).  That's why I realized simple structure for work easy with hash tables.